### PR TITLE
chore(ci): Update node version for telemetry check

### PR DESCRIPTION
**Pull request by Josh-Walker-GM**
**Problem**
After updating the engine check requirements to a minimum of `16.20` the GitHub ci node version is too low for telemetry checks to run as intended.

**Changes**
Update from `16` to `16.20.0` in the node version.

### CODE REVIEW

**Comment by Josh-Walker-GM** [comment]
I'm going to enable auto-merge on this without an additional approval given how minor the change is and that it'll be blocking CI on all other PRs until this is corrected.

File: .github/workflows/ci.yml
Comment (jtoar) [inline .github/workflows/ci.yml:333]: This seems like a miss on `actions/setup-node`'s part. Not sure why 16 doesn't pull down the latest.

Just checked their docs—we could set the "check-latest" option to true:

```suggestion
        node-version: [16, 18]
        check-latest: true
```

https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#check-latest-version

But maybe it's better to be more exact like you have it here since we support a minimum 16 version? I'll leave it up to you! If we go with `16.20.0` I'll have to change the settings to accept a check passing with that title, which is no big deal. Docs for reference:

https://github.com/redwoodjs/redwood/blob/385064dbc5a8c775e69a837277722a5509ef21ca/packages/create-redwood-app/template/package.json#L17

**Review by jtoar — APPROVED** [review]

File: .github/workflows/ci.yml
Comment (jtoar) [reply .github/workflows/ci.yml:333]: I just changed the check name for now

**Pull request merged**
